ntbs-service-unit-tests/DataMigration/NotificationMapperTest.cs

```diff
@@ -673,6 +679,124 @@ public async Task correctlyMaps_TravelAndVisitorNumbersToNullWhenNoCountriesAdde
+            Assert.Equal("Some other immunosuppression", notification.ImmunosuppressionDetails.OtherDescription);
+        }
+
+        [Fact]
+        public async Task correctlyMaps_ImmunosuppressionDetails_WithoutDescriptionWhenHasOtherIsTrue()
+        {
+            // Arrange
+            const int runId = 12345;
+            var legacyIds = new List<string> { "132502" };
+            SetupNotificationsInGroups(("132502", "12"));
+
+            const string royalBerkshireCode = "TBS001";
+            _hospitalToTbServiceCodeDict = new Dictionary<Guid, TBService>
+            {
+                {new Guid("B8AA918D-233F-4C41-B9AE-BE8A8DC8BE7A"), new TBService {Code = royalBerkshireCode}},
+            };
+
+            // Act
+            var notification = (await _notificationMapper.GetNotificationsGroupedByPatient(null,
+                    runId,
+                    legacyIds))
+                .SelectMany(group => group)
+                .Single();
+
+            // Assert
+            Assert.Equal(Status.Yes, notification.ImmunosuppressionDetails.Status);
+            Assert.False(notification.ImmunosuppressionDetails.HasBioTherapy);
+            Assert.False(notification.ImmunosuppressionDetails.HasTransplantation);
+            Assert.True(notification.ImmunosuppressionDetails.HasOther);
+            Assert.Equal("No description provided in the legacy system",
+                notification.ImmunosuppressionDetails.OtherDescription);
+        }
+
+        [Fact]
+        public async Task correctlyMaps_ImmunosuppressionDetails_WhenAllTypesAreFalse()
+        {
+            // Arrange
+            const int runId = 12345;
+            var legacyIds = new List<string> { "132503" };
+            SetupNotificationsInGroups(("132503", "12"));
+
+            const string royalBerkshireCode = "TBS001";
+            _hospitalToTbServiceCodeDict = new Dictionary<Guid, TBService>
+            {
+                {new Guid("B8AA918D-233F-4C41-B9AE-BE8A8DC8BE7A"), new TBService {Code = royalBerkshireCode}},
+            };
+
+            // Act
+            var notification = (await _notificationMapper.GetNotificationsGroupedByPatient(null,
+                    runId,
+                    legacyIds))
+                .SelectMany(group => group)
+                .Single();
+
+            // Assert
+            Assert.Equal(Status.Yes, notification.ImmunosuppressionDetails.Status);
+            Assert.False(notification.ImmunosuppressionDetails.HasBioTherapy);
+            Assert.False(notification.ImmunosuppressionDetails.HasTransplantation);
+            Assert.True(notification.ImmunosuppressionDetails.HasOther);
+            Assert.Equal("No immunosuppression type was provided in the legacy record",
+                notification.ImmunosuppressionDetails.OtherDescription);
+        }
+
         // Data for this has been taken from an edited test notification, relating to NTBS-2478
```

Comment:
Do we have a test for stripping the invalid characters at the moment?

Reply:
We do now that I've added one!